apps/api/src/app/notification-groups/usecases/update-notification-group/update-notification-group.usecase.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { NotificationGroupRepository, NotificationGroupEntity } from '@novu/dal';
+import { UpdateNotificationGroupCommand } from './update-notification-group.command';
+import { NotificationGroupResponseDto } from '../../dtos/notification-group-response.dto';
+
+@Injectable()
+export class UpdateNotificationGroup {
+  constructor(private notificationGroupRepository: NotificationGroupRepository) {}
+
+  async execute(command: UpdateNotificationGroupCommand) {
+    const { id, environmentId, name } = command;
+
+    const item = await this.notificationGroupRepository.findOne({
+      _environmentId: environmentId,
+      _id: id,
+    });
+
+    if (!item) {
+      throw new NotFoundException();
+    }
+
+    await await this.notificationGroupRepository.update(
+      {
+        _id: id,
+        _environmentId: environmentId,
+      },
+      {
+        $set: {
+          name,
+        },
+      }
+    );
+
+    const result = await this.notificationGroupRepository.findOne({
+      _environmentId: environmentId,
+      _id: id,
+    });
+
+    if (!result) {
+      throw new NotFoundException(`Notification group with id ${id} not found`);
+    }
```

Comment:
Instead of doing this you can check inside the NotificationGroupRepository if the response from the update method returns that an entry has been changed and manage the logic accordingly.
Also it feels a bit weird that you check again if the group exists after updating it when you have already checked it before updating it. 🤔

Reply:
I need to return an object of the update code as per the `NotificationGroupResponseDto`